File: Java/Tests/jdi-uitest-wintests/src/test/java/com/epam/jdi/uitests/testing/unittests/tests/common/LabelTests.java
package com.epam.jdi.uitests.testing.unittests.tests.common;

import com.epam.jdi.uitests.core.interfaces.common.ILabel;
import com.epam.jdi.uitests.testing.unittests.InitTests;
import org.testng.annotations.BeforeMethod;
import org.testng.annotations.Factory;
import org.testng.annotations.Test;

import java.lang.reflect.Method;
import java.util.function.Supplier;

import static com.epam.jdi.uitests.core.preconditions.PreconditionsState.isInState;
import static com.epam.jdi.uitests.testing.unittests.enums.Preconditions.METALS_AND_COLORS_PAGE;
import static com.epam.jdi.uitests.testing.unittests.pageobjects.EpamJDIApplication.metalsColorsPage;
import static com.epam.jdi.uitests.testing.unittests.tests.complex.CommonActionsData.checkCalculate;

/**
 * Created by Dmitry_Lebedev1 on 15/12/2015.
 */
public class LabelTests extends InitTests {

    private Supplier<ILabel> label = () -> metalsColorsPage.calculate;

    @BeforeMethod
    public void before(final Method method) {
        isInState(METALS_AND_COLORS_PAGE, method);
    }

    @Test
    public void clickTest() {
        label.get().click();
        checkCalculate("Summary: 3");
    }

    @Factory
    public Object[] textTests() {
        return new Object[]{
                new TextTests("Label", METALS_AND_COLORS_PAGE, label::get, "CALCULATE", "CUL", ".*LCU.*")
        };
    }
}

File: Python/Test/jdi_uitests_webtests/test/common/label_tests.py
from JDI.jdi_assert.testing.assertion import Assert
from Test.jdi_uitests_webtests.main.enums.preconditions import Preconditions
from Test.jdi_uitests_webtests.main.page_objects.epam_jdi_site import EpamJDISite
from Test.jdi_uitests_webtests.test.init_tests import InitTests


class LabelTests(InitTests):
    label = EpamJDISite.metals_colors_page.calculate_label

    def setUp(self):
        Preconditions.METALS_AND_COLORS_PAGE.is_in_state()

    def test_click(self):
        self.label.click()
        Assert.assert_element_test(self.label, "CALCULATE")
